lib/matplotlib/colorbar.py

```diff
@@ -1276,10 +1268,18 @@ def update_bruteforce(self, mappable):
         self.formatter = None
 
         # clearing the axes will delete outline, patch, solids, and lines:
-        self.outline = None
-        self.patch = None
+        self.outline = mpatches.Polygon(
+            np.empty((0, 2)),
+            edgecolor=mpl.rcParams['axes.edgecolor'], facecolor='none',
+            linewidth=mpl.rcParams['axes.linewidth'], closed=True, zorder=2)
+        self.ax.add_artist(self.outline)
+        self.outline.set(clip_box=None, clip_path=None)
+        self.patch = mpatches.Polygon(
+            np.empty((0, 2)),
+            color=mpl.rcParams['axes.facecolor'], linewidth=0.01, zorder=-1)
+        self.ax.add_artist(self.patch)
```

Comment:
This code is repeated from `__init__`. Should this be moved to a private helper to remove duplication? Maybe only returning the outline and patch without adding them to ax within the helper?

Reply:
this is getting deprecated in https://github.com/matplotlib/matplotlib/pull/15982.